retrieval/targetmanager.go

```diff
@@ -113,7 +113,12 @@ type targetGroupUpdate struct {
 func (tm *TargetManager) Run() {
 	log.Info("Starting target manager...")
 
-	tm.done = make(chan struct{})
+	// It's important that we use a local copy of the "done" channel to listen
```

Comment:
It's not so much a copy than a reference, no? Alternatively you could also just explicitly pass the channel to he defer func.